apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/Trace/SimpleManagedTracerTest.cs

```diff
@@ -364,19 +364,29 @@ public void MultipleSpans()
             mockConsumer.Setup(c => c.Receive(Match.Create(rootMatcher)));
 
             var childThreadsReleased = new ManualResetEventSlim(initialState: false);
+            var startedChildSpans = 0;
 
             Func<string, string, Task> op = async (childName, grandchildName) =>
             {
                 childThreadsReleased.Wait();
 
                 using (tracer.StartSpan(childName))
                 {
+                    Interlocked.Increment(ref startedChildSpans);
+
                     await Task.Yield();
                     using (tracer.StartSpan(grandchildName))
                     {
                         await Task.Yield();
                     }
-                    await Task.Yield();
+
+                    do
+                    {
+                        await Task.Yield();
+                        // Don't allow the span to close until both are started
+                        // to make sure we only get a single Receive call below
+                        // instead of two.
+                    } while (startedChildSpans < 2);
```

Comment:
I suspect this should probably be something also using Interlocked to ensure it actually reads from main memory, in theory... but unless we see hanging tests, I'm happy to assume that there's enough going on in async/await to make the JIT compiler not optimize the read away.